File: 6TH_SEMESTER/SWE 4604 - SOFTWARE TESTING AND QUALITY ASSURANCE LAB/6/testinglab06/src/tests/HomePage.java
package tests;

import org.junit.jupiter.api.Test;
import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.interactions.Actions;

//210042112

public class HomePage extends BaseTest {

    @Test
    public void Sorting()
    {
        WebElement dropdown = driver.findElement(By.cssSelector("select[data-test='sort']"));
        dropdown.click();


        WebElement option = driver.findElement(By.cssSelector("option[value='price,desc']"));
        option.click();
    }

    @Test
    public void SearchBar()
    {
        WebElement textBox = driver.findElement(By.cssSelector("input[data-test='search-query']"));
        textBox.sendKeys("tool cabinet");

        WebElement searchButton = driver.findElement(By.cssSelector("button[data-test='search-submit']"));
        searchButton.click();

    }

    @Test
    public void FilterCheckBoxByCategory()
    {

        WebElement hammerCheckbox = driver.findElement(By.xpath("//label[contains(., 'Hammer')]/input[@type='checkbox']"));
        if (!hammerCheckbox.isSelected())
        {
            hammerCheckbox.click();
        }
    }


    @Test
    public void FilterCheckBoxByBrand()
    {
        WebElement forgeflexCheckbox = driver.findElement(By.xpath("//label[contains(., ' ForgeFlex Tools')]/input[@type='checkbox']"));
        if (!forgeflexCheckbox.isSelected())
        {
            forgeflexCheckbox.click();
        }

    }

    @Test
    public void PriceRange()
    {
        WebElement minelement = driver.findElement(By.cssSelector(".ngx-slider-pointer-min"));
        Actions minbuilder = new Actions(driver);
        minbuilder.moveToElement(minelement).clickAndHold().perform();
        minelement.click();

        WebElement maxelement = driver.findElement(By.cssSelector(".ngx-slider-pointer-max"));
        Actions maxbuilder = new Actions(driver);
        maxbuilder.moveToElement(maxelement).clickAndHold().perform();
        maxelement.click();
    }

}

// -_- N4M154 -_-

File: 6TH_SEMESTER/SWE 4604 - SOFTWARE TESTING AND QUALITY ASSURANCE LAB/6/Lab06CSharp/TestProject1/HomePage.cs
using OpenQA.Selenium;
using OpenQA.Selenium.Interactions;

namespace Lab06
{
    public class HomePage : BaseTest
    {
        [Fact]
        public void Sorting()
        {
            var dropdown = driver.FindElement(By.CssSelector("select[data-test='sort']"));
            dropdown.Click();


            var option = driver.FindElement(By.CssSelector("option[value='price,desc']"));
            option.Click();
        }

        [Fact]
        public void SearchBar()
        {
            var textBox = driver.FindElement(By.CssSelector("input[data-test='search-query']"));
            textBox.SendKeys("tool cabinet");

            var searchButton = driver.FindElement(By.CssSelector("button[data-test='search-submit']"));
            searchButton.Click();

        }

        [Fact]
        public void FilterCheckBoxByCategory()
        {
            //hammer
            var hammerCheckbox = driver.FindElement(By.XPath("//div[@class='checkbox']/label[input[@name='category_id' and @value='01K0H5EZJQ94YSYKZ59TFN2NTR']]//input"));
            if (!hammerCheckbox.Selected)
            {
                hammerCheckbox.Click();
            }

        }

        [Fact]
        public void FilterCheckBoxByBrand()
        {
            var forgeflexCheckbox = driver.FindElement(By.XPath("//div[@class='checkbox']/label[input[@name='brand_id' and @value='01K0H5EZ4J4H02FSM090Y4PDAT']]//input"));
            if (!forgeflexCheckbox.Selected)
            {
                forgeflexCheckbox.Click();
            }

        }

        [Fact]
        public void PriceRange()
        {
            var minelement = driver.FindElement(By.CssSelector(".ngx-slider-pointer-min"));
            Actions minbuilder = new Actions(driver);
            minbuilder.MoveToElement(minelement).ClickAndHold().Perform();
            minelement.Click();

            var maxelement = driver.FindElement(By.CssSelector(".ngx-slider-pointer-max"));
            Actions maxbuilder = new Actions(driver);
            maxbuilder.MoveToElement(maxelement).ClickAndHold().Perform();
            maxelement.Click();
        }



    }
}
